crates/composefs-setup-root/src/main.rs

```diff
@@ -198,12 +198,19 @@ fn mount_subdir(
 }
 
 // Implementation
-fn parse_composefs_cmdline<H: FsVerityHashValue>(cmdline: &str) -> Result<H> {
-    let Some(digest) = get_cmdline_value(cmdline, "composefs=") else {
+fn parse_composefs_cmdline<H: FsVerityHashValue>(cmdline: &str) -> Result<(H, bool)> {
+    let Some(mut digest) = get_cmdline_value(cmdline, "composefs=") else {
         bail!("Unable to find composefs= cmdline parameter");
     };
 
-    H::from_hex(digest).context("Parsing composefs=")
+    let mut insecure = false;
+    if let Some(stripped) = digest.strip_prefix('?') {
```

Comment:
I think we were going to move this code into `cmdline.rs` in the `composefs-boot` crate?  We might also use `~` instead of `?`.